### PR TITLE
Add Connection type

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
The Connection type will be used for managing connections to ARM.
Updated documentation with examples.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

File: sdk/armcore/connection.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+func NewConnection(endpoint string, cred azcore.TokenCredential, options *ConnectionOptions) *Connection {
+	if options == nil {
+		o := DefaultConnectionOptions()
+		options = &o
+	}
+	policies := []azcore.Policy{
+		azcore.NewTelemetryPolicy(&options.Telemetry),
+	}
+	policies = append(policies, NewRPRegistrationPolicy(cred, &options.RegisterRPOptions))
+	policies = append(policies,
```
Comment (catalinaperalta) [inline sdk/armcore/connection.go:61]: Why is this in two separate appends?

File: sdk/armcore/connection.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	return &Connection{u: endpoint, p: p}
+}
+
+// Do invokes the Do() method on the connection's pipeline.
+func (c *Connection) Do(req *azcore.Request) (*azcore.Response, error) {
+	return c.p.Do(req)
+}
+
+// Endpoint returns the connection's ARM endpoint.
+func (c *Connection) Endpoint() string {
```
Comment (catalinaperalta) [inline sdk/armcore/connection.go:81]: Trying to remember where we need this in the generated code?

Comment (catalinaperalta) [reply sdk/armcore/connection.go:81]: Nvm when creating the request :P

File: sdk/armcore/connection_test.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/mock"
+	//"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	//"github.com/Azure/azure-sdk-for-go/sdk/internal/mock"
```
Comment (catalinaperalta) [inline sdk/armcore/connection_test.go:17]: I'm thinking these should be removed?

**Review by catalinaperalta — APPROVED** [review]
LGTM, left a couple of comments to address.

File: sdk/armcore/connection.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/armcore/connection.go:61]: This is an artifact from copying the generated code, will fix.

Comment (jhendrixMSFT) [reply sdk/armcore/connection.go]: Fixed

File: sdk/armcore/connection_test.go (hunk 3 above)
Comment (jhendrixMSFT) [reply sdk/armcore/connection_test.go]: Fixed

**Pull request merged**